spec/controllers/projects_controller_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe ProjectsController, type: :controller do
+  before(:each) do
+    @user = User.create!(name: 'midd', email: 'user@example.org', password: 'very-secret')
+    @user.role = :developer
+    @user.save!
+  end
+  context 'get' do
+    it 'returns a successful response' do
+      get :index
+      expect(response).to be_successful
+    end
+
+    it 'assigns @projects' do
+      @project = Project.create(title: 'Project1',
+                                info: 'body project',
+                                user: User.new(email: 'midda999@mail.ru',
+                                               name: 'Anton',
+                                               password: '123456'))
+      get :index
+      expect(assigns(:projects)).to eq([@project])
+    end
+
+    it 'project find by id' do
+      @project = Project.new(title: 'Project1', info: 'info to project 1', user: @user)
+      @project.save!
+      get :show, params: { id: @project.id }
+      expect(assigns(:project)).to eq(@project)
+    end
+  end
+
+  context 'get template' do
+    it 'index' do
+      get :index
+      expect(response).to render_template('index')
+    end
+
+    it 'new' do
+      sign_in @user
+      get :new
+      expect(response).to render_template('new')
+    end
+  end
+
+  context 'project edit by user' do
+    it 'author' do
+      @project = Project.new(title: 'Project1', info: 'info to project 1', user: @user)
+      @project.save!
+      get :edit, params: { id: @user.projects.first }
+      expect(@user.projects.first).to eq(@project)
+    end
+
+    it 'not author' do
+      sign_in @user
+      user2 = User.create!(name: 'midd22', email: 'usetrr@example.org', password: 'ver12y-secret')
```

Comment:
Again, can we unify user and project creation in before and do minimal setup inside unit tests?